x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/action/TransportPutQueryRuleAction.java

```diff
@@ -43,7 +43,7 @@ public TransportPutQueryRuleAction(
     protected void doExecute(Task task, PutQueryRuleAction.Request request, ActionListener<PutQueryRuleAction.Response> listener) {
         String queryRulesetId = request.queryRulesetId();
         QueryRule queryRule = request.queryRule();
-        systemIndexService.putQueryRule(queryRulesetId, queryRule, listener.map(r -> new PutQueryRuleAction.Response(r.getResult())));
+        systemIndexService.putQueryRule(queryRulesetId, queryRule, ActionListener.wrap(listener::onResponse, listener::onFailure));
```

Comment:
Dumb question: how is this different than passing `listener` directly?

Reply:
I had to do this because we're modifying the response from a `DocWriteResponse` to ` PutQueryRuleAction.Response` but it's basically just wrapping it so we can change it to created for rulesets with new rules.

Reply:
PS there is no such thing as a dumb question 😛